### PR TITLE
implemented ui for auh page

### PR DESCRIPTION
**Pull request by Maxim-Gumeniuk**

### CODE REVIEW

**Comment by Maxim-Gumeniuk** [comment]
[sergienko339@gmail.com](mailto:sergienko339@gmail.com)

File: src/ui-bricks/auth/form.tsx [hunk 1]
```diff
@@ -40,29 +40,55 @@ export const AuthForm: FC<Props> = memo(({ inputArray }) => {
...
-                                    forget a password ?
-                                </span>
+                                {inputArray.map(
+                                    ({
+                                        type,
+                                        placeholder,
+                                        value,
+                                        change,
+                                        errors,
+                                        touched,
```
Comment (SpaceHolderdJs) [inline src/ui-bricks/auth/form.tsx:51]: `input`

File: src/ui-bricks/auth/form.tsx [hunk 2]
```diff
@@ -40,29 +40,55 @@ export const AuthForm: FC<Props> = memo(({ inputArray }) => {
...
+                                        return (
+                                            <Fragment key={type}>
+                                                <input
+                                                    id={type}
+                                                    name={type}
+                                                    type={type}
+                                                    className="rounded-lg bg-grey p-4 placeholder-uppercase"
+                                                    placeholder={placeholder}
+                                                    value={value}
+                                                    onChange={change}
```
Comment (SpaceHolderdJs) [inline src/ui-bricks/auth/form.tsx:62]: <input {...input} id={input.type} type={input.type} />

**Review by SpaceHolderdJs — COMMENTED** [review]
Looks great

**Comment by Maxim-Gumeniuk** [comment]
[sergienko339@gmail.com](mailto:sergienko339@gmail.com)

File: src/app/providers/index.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+"use client";
+
+import { SnackbarProvider } from "notistack";
+import { ReactNode } from "react";
+
+type Props = {
+    children: ReactNode;
+};
+
+export function Providers({ children }: Props) {
```
Comment (SpaceHolderdJs) [inline src/app/providers/index.tsx:10]: `FC<PropsWithChildren>`

File: src/api/auth/index.ts [hunk 4]
```diff
@@ -11,9 +12,10 @@ const registration = async (param: Icredentials) => {
             email,
             password
         })
-    } catch(e) {
-        console.log(e);
-        
+    } catch(error: any) {
```
Comment (SpaceHolderdJs) [inline src/api/auth/index.ts:15]: export type GenericError = `<T>` =  T | typeof  Error

File: src/api/auth/index.ts [hunk 5]
```diff
@@ -11,9 +12,10 @@ const registration = async (param: Icredentials) => {
...
+        console.log(error.response.data);
```
Comment (SpaceHolderdJs) [inline src/api/auth/index.ts:16]: remove

**Pull request merged**